src/background/host.ts

```diff
@@ -1,29 +1,31 @@
 import Peer from "peerjs";
-import { Message, MessageType } from "../Message";
+import { Friend, IFriend } from "../Friend";
+import { IFriendMessage, Message, MessageType, IPollMessage } from "../Message";
 
 export class Host {
-    #port: chrome.runtime.Port;
-    #peer = new Peer();
-    #friends = new Set<Peer.DataConnection>();
-
+    /** Whether or not the current Host instance is initialized */
     public isReady = false;
+
+    /** The current id */
     public id: string | null = null;
-    // eslint-disable-next-line @typescript-eslint/no-empty-function
-    public onChangeFriends: (count: number, delta: number) => void = () => {};
 
-    #handleConnect = (conn: Peer.DataConnection): void => {
-        this.#friends.add(conn);
-        this.onChangeFriends(this.#friends.size, 1);
+    /** The current user's chosen displayName */
+    public displayName = "";
 
-        conn.on("data", (data: Message) => {
-            this.#port.postMessage(data);
-        });
+    /** See: https://developer.chrome.com/extensions/runtime#type-Port */
+    #port: chrome.runtime.Port;
 
-        conn.on("close", () => {
-            this.#friends.delete(conn);
-            this.onChangeFriends(this.#friends.size, -1);
-        });
-    };
+    /** The underlying peerjs Peer instance */
+    #peer = new Peer();
+
+    /** Map of friend ids to friends  */
+    #friends = new Map<string, Friend>();
+
+    /** List of friend change subscribers */
+    private friendChangeSubscribers: ((count: number, delta: number) => void)[] = [];
```

Comment:
At this point we might want to just import an NPM module for `EventEmitter` and use all those pre-built goodies.

Reply:
I'm hesitant to do that since we don't really need 99% of the functionality that comes with those packages and they're usually not very small. Should I revert this change to the callback that we set somewhere else?

Reply:
I'd vote to put this back as just a `readwrite` function as on `master`, since you always have exactly one subscriber and it's the same spot that it was before.